compiler/rustc_next_trait_solver/src/solve/trait_goals.rs

```diff
@@ -87,6 +87,19 @@ where
                 .map(|pred| goal.with(cx, pred));
             ecx.add_goals(GoalSource::ImplWhereBound, where_clause_bounds);
 
+            // We currently elaborate all supertrait obligations from impls. This
+            // can be removed when we actually do coinduction correctly and just
+            // register that the impl header must be WF.
```

Comment:
this comment doesn't feel right?

Reply:
I guess it should be "actually do coinduction correctly and just register supertrait obligations always". The impl header WF check is a totally different approach, yeah.

Reply:
We also don't elaborate all supertrait obligations :sweat_smile: we only elaborate region obligations as elaborating all of them causes inductive solver cycles

Reply:
Well I think the point here is that we do deep elaboration `elaborate(...)` rather than shallow elaboration `tcx.explicit_supertraits(...)`.  I'll reword the comment.